extensions/amp-mraid/0.1/mraid-service.js

```diff
@@ -86,9 +87,19 @@ export class MraidService {
         (exposedPercentage,
           visibleRectangle,
           unusedOcclusionRectangles) => {
-          callback({visibleRect: visibleRectangle,
-            visibleRatio: exposedPercentage});
+          // AMP handles intersectionRatio from 0.0 to 1.0
+          callback({
```

Comment:
I think we need to be a bit more defensive here against input that's not what we're expecting.  When exposedPercentage is 0 I think visibleRectangle will be null.  What about:

```
            visibleRect: layoutRectLtwh(
                visibleRectangle ? visibleRectangle.x : 0,
                visibleRectangle ? visibleRectangle.y : 0,
                visibleRectangle ? visibleRectangle.width : 0,
                visibleRectangle ? visibleRectangle.height : 0),
```

Reply:
Thank you @jeffkaufman. That's exactly what I found, visibleRectangle is null when the ratio is 0.

Reply:
Also I think it's worth considering how to support future host API integration works. Converting visibleRectangle to AMP supported layoutRect can be a bit tricky. I personally spent some time debugging the logic here myself, so I want to make future integration is smooth to whoever is less familiar with the AMP runtime.